public/app/features/plugins/admin/helpers.ts

```diff
@@ -49,8 +49,15 @@ export function mergeLocalsAndRemotes({
       if (configCore.featureToggles.managedPluginsInstall && config.pluginAdminExternalManageEnabled) {
         catalogPlugin.isFullyInstalled = catalogPlugin.isCore
           ? true
-          : instancesSet.has(remotePlugin.slug) && catalogPlugin.isInstalled;
-        catalogPlugin.isInstalled = instancesSet.has(remotePlugin.slug) || catalogPlugin.isInstalled;
+          : instancesMap.has(remotePlugin.slug) && catalogPlugin.isInstalled;
+
+        catalogPlugin.isInstalled = instancesMap.has(remotePlugin.slug) || catalogPlugin.isInstalled;
+
+        const instancePlugin = instancesMap.get(remotePlugin.slug);
+        catalogPlugin.isUpdating =
+          instancesMap.has(remotePlugin.slug) &&
+          catalogPlugin.hasUpdate &&
+          catalogPlugin.installedVersion !== instancePlugin?.version;
```

Comment:
I think we should make the name `isUpdating` a bit more specific / clear, as if I understand correctly it is only used for checking if the plugin is being updated _in cloud_ (in an on-prem instance this is going to be `false` even if the plugin is being updated), am I right?

Reply:
Exactly, I think the same comment that you did [here](https://github.com/grafana/grafana/pull/81907#discussion_r1482553436) applies for this case.

I will change the variable name accordingly. :smile: